proposals/lip-introduce_unlocking_condition.md

```diff
@@ -0,0 +1,111 @@
+In particular, we want to prevent a group of delegates from leaving the set of active delegates, but refusing to sign a certificate that attests this change.
+We want to achieve this by only letting a delegate and its voters unlock, after the delegate was voted out of the active set, if a certificate is generated for the last height where the delegate was active.
+This certificate will attest that the respective delegate is no longer part of the active set of delegates of the chain.
+
+![Example of an unvote transaction](lip-introduce_unlocking_condition/example_unvote.png)
+
+*Figure 1: The result of an unvote transaction `t` included in round `r` only comes into effect in round `r+3` as the vote snapshots for delegate selection are taken from two rounds ago.*
+
+More concretely, consider one unvote transaction `t` included at height `h` in round `r` that results in a delegate `d` leaving the set of active delegates, see also the figure above as illustration.
+The vote snapshots used for selecting the active delegates at the beginning of a round are taken from two rounds ago.
+This means that the last block `b` of round `r+2` is the last block when delegate `d` is active.
+In particular, the `validatorsHash` property in block `b` attests a validator set not including `d`.
+As certificates are generated for all end-of-round blocks, we require that the certificate signature for the certificate derived from block `b` is included in the chain, before the account submitting the unvote transaction can unlock.
+This means that the following inequality has to hold for the unlock included at height `h` to be valid:
+
+```
+lastHeightOfRound(h + 2 * ROUND_LENGTH) <= maxHeightCertified.
+```
+
+Here
+- `ROUND_LENGTH` is the round length in blockchains using the [DPoS module LIP][dpos-module-lip],
+- `lastHeightRound()` returns the last height of the round containing the given height,
+- `maxHeightCertified` is the maximum height of a block for which the chain contains a certificate.
+
+## Specification
+
+### Notation
+
+- We use the constant `ROUND_LENGTH` defined in the [LIP "Define state and state transitions of DPoS module"][dpos-module-lip] to denote the round length in the DPoS chain, i.e., the number of active plus randomly selected standby delegates.
+- We write `bftModule.fct` for a call to the function `fct` defined in the [BFT module][bft-module-lip].
+
+### Internal Functions
+
+#### lastHeightOfRound
+
+ For a height `h` of round `r` as input, this function returns the height of the last block of round `r`. In particular, if `h` is the last height of a round, we have `h = lastHeightOfRound(h)`.
+
+```python
+lastHeightOfRound(h):
+    r = roundNumber(h)
+    return r * ROUND_LENGTH + genesisHeight
+```
+
+Here `roundNumber` is a function defined in the [LIP "Define state and state transitions of DPoS module"][dpos-module-lip] and `genesisHeight` denotes the height of the genesis block as stored in the [DPoS module][dpos-module-lip].
```

Comment:
Consider linking directly to the function when updating the links (e.g., link https://research.lisk.com/t/define-state-and-state-transitions-of-dpos-module/320#roundnumber-63 for now)

Reply:
Thanks. I updated the link to link directly to the specific function.